src/components/app/App.tsx

```diff
@@ -0,0 +1,113 @@
+import { ReactElement, useState, useEffect } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { PAGE_TITLE, PAGE_NUMBER_DEFAULT, getAuthors } from '../../utils/data';
+import { httpGet } from '../../utils/httpGet';
+import Sidebar from '../sidebar/Sidebar';
+import Results from '../results/Results';
+import Spinner from '../spinner/Spinner';
+import styles from './App.module.scss';
+import type { SectionsResponseResults, SectionsResponse, SearchResponse, Author } from '../../types/types';
+
+function updateSearchParams(searchParams: URLSearchParams, params: Record<string, string | null>): void {
+    Object.entries(params).forEach((value) => {
+        if (value[1] === '') {
+            searchParams.delete(`${value[0]}`)
```

Comment:
Is deleting those needed? If you set `q` or `section` to empty string the results are not the same like in case of not having those defined at all?